### DISCUSSION TITLE
Feature to Add <link rel="next/prev" href="..." > in head.

### DISCUSSION THREAD

**Discussion by umang-goyal** [vercel/next.js#52923] [App Router] [answered]
Unable to find how to add <link rel="next"> or <link rel="prev> in Metadata. 
https://nextjs.org/docs/app/building-your-application/optimizing/metadata

**Comment by matteopaoli** [comment]


**Reply by insanity54** [reply]
@matteopaoli  I don't think that answers the question. putting values inside metadata.other creates a `<meta>` tag. The OP (and I) are asking for how to create a `<link>` tag.

From the docs

> Next.js has a Metadata API that can be used to define your application metadata (e.g. meta and link tags inside your HTML head element) for improved SEO and web shareability.

The docs do not mention anything further regarding link tags.

**Comment by insanity54** [comment]
I was looking to do this on my next.js 13 app, so visiting web browsers were made aware of my RSS feed. To do this, I needed to put a `<link>` tag inside `<head>`.

From browsing [discussions](https://github.com/vercel/next.js/discussions/52926), it looks like this isn't supported using a [statically defined Metadata object](https://nextjs.org/docs/app/api-reference/functions/generate-metadata#metadata-object). 

Here's what I did to get this to work.

```tsx
import { ReactNode } from 'react'
import Footer from "./components/footer"
import Navbar from "./components/navbar"
import "../assets/styles/global.sass";
import "@fortawesome/fontawesome-svg-core/styles.css";
import { AuthProvider } from './components/auth'
import type { Metadata } from 'next'

export const metadata: Metadata = {
  title: 'example.com,
  description: "The Galaxy's Best Coffee",
  other: {
    RATING: 'RTA-5042-1996-1400-1577-RTA'
  }
}

type Props = {
  children: ReactNode;
}


export default function RootLayout({
  children,
}: Props) {
  return (
    <html lang="en">
      <head>
        <link rel="alternate" type="application/atom+xml" title="ATOM" href="/feed/feed.xml" />
        <link rel="alternate" type="application/rss+xml" title="RSS" href="/feed/feed.rss" />
        <link rel="alternate" type="application/json" title="JSON" href="/feed/feed.json" />
      </head>
      <body>
        <AuthProvider>
          <Navbar />
          <div className="container">
            {children}
            <Footer />
          </div>
        </AuthProvider>
      </body>
    </html>
  )
}
```

In the above code, I defined my own `<head>` tag with the necessary `<link>` tags. Next must merge this tag with the metadata, because I'm seeing all the `<meta>` and `<link>` tags in the rendered output.

**Comment by matteopaoli** [comment, marked as answer]
I've just found out there is a workaround which consist in injecting custom `<link />` tags using the `icons` object in the Metadata API

```
export const metadata: Metadata = {
  title: 'example.com,
  description: "The Galaxy's Best Coffee",
  icons: {
    other: [
     { rel: 'next', url: 'https://example.com/search?page=3' },
     { rel: 'prev', url: 'https://example.com/search?page=1' },
    ]
  }
}
```

I've just inserted this on my website and seems to work correctly. 
This is the output

```
<link rel="next" href="https://example.com/search?page=3">
<link rel="prev" href="https://example.com/search?page=1">
```

**Reply by kirstymullen** [reply]
Having tried this workaround, it's worth noting that it overrides the file-based API for your actual icons (which is the recommended approach) so they no longer work.

**Reply by ung0v** [reply]
> Having tried this workaround, it's worth noting that it overrides the file-based API for your actual icons (which is the recommended approach), so they no longer work.

You can get the parent metadata: https://nextjs.org/docs/app/api-reference/functions/generate-metadata#generatemetadata-function
Here's an example 
```
export async function generateMetadata(
  { params, searchParams }: Props,
  parent: ResolvingMetadata
): Promise<Metadata> {
 return {
  title: 'example.com,
  description: "The Galaxy's Best Coffee",
  icons: {
    other: [
     ...(await parent).icons,
     { rel: 'next', url: 'https://example.com/search?page=3' },
     { rel: 'prev', url: 'https://example.com/search?page=1' },
    ]
  }
 }
}
```

**Reply by yandodov** [reply]
As I wrote [here](https://github.com/vercel/next.js/discussions/52923#discussioncomment-12611243), this is now natively supported in Next 15.2.

**Reply by Livideakoto** [reply]
You're genius man! 🥳

**Comment by mylesthedev** [comment]
This seems like it could have issues down he track nor does it seems appropriate to put SEO tags linke next/prev under icons?

**Reply by mylesthedev** [reply]
In saying that... Google doesn't even recommend using these links for indexing any longer, soo probably no point adding support.

**Reply by kirstymullen** [reply]
Google doesn't utilise them anymore, but other search engines and tools do e.g. Bing.

I'm adding these conditionally in `layout.tsx` for now. I'd prefer a cleaner solution down the line for these and other `link` tags.

**Reply by Saul-BT** [reply]
Any updates on this? A cleaner alternative over use `metadata.icons.other`? 😿

**Comment by yandodov** [comment]
FYI, the `prev` and `next` tags are now natively supported in Next 15.2, thanks to @Zach-Jaensch.

- Blog post: https://nextjs.org/blog/next-15-2#other-changes
- PR: https://github.com/vercel/next.js/pull/74737

Example:

```ts
import { Metadata } from "next";

export async function generateMetadata(): Promise<Metadata | undefined> {
  return {
    // ...
    pagination: {
      previous: "blog/4",
      next: "blog/5",
    },
  };
}
```

**Reply by endrits079** [reply]
any chance adding this to 14?
